backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/EdgeComponent.java

```diff
@@ -135,6 +139,26 @@ public Element render() {
         return new Fragment(fragmentProps);
     }
 
+    private Label getLabel(VariableManager labelVariableManager, Optional<Edge> optionalPreviousEdge, Function<VariableManager, Optional<Label>> labelProvider, Function<Edge, Label> labelGetter,
+            LabelBoundsProvider labelBoundsProvider) {
+        Label label = labelProvider.apply(labelVariableManager).orElse(null);
+        if (label != null) {
+            Optional<Label> optionalPreviousLabel = optionalPreviousEdge.map(labelGetter);
+            label = this.getLayoutedLabel(label, optionalPreviousLabel, labelBoundsProvider);
+        }
+        return label;
+    }
+
+    private Label getLayoutedLabel(Label label, Optional<Label> optionalPreviousLabel, LabelBoundsProvider labelBoundsProvider) {
+        //@formatter:off
+       return Label.newLabel(label)
+                .position(optionalPreviousLabel.map(Label::getPosition).orElseGet(() -> labelBoundsProvider.getPosition()))
+                .size(optionalPreviousLabel.map(Label::getSize).orElseGet(() -> labelBoundsProvider.getSize()))
+                .alignment(optionalPreviousLabel.map(Label::getAlignment).orElseGet(() -> labelBoundsProvider.getAlignment()))
+                .build();
+       //@formatter:on
+    }
+
```

Comment:
It makes no sense to create every labels twice, instead change the following APIs which should never have existed:

- EdgeDescription#getBeginLabelProvider
- EdgeDescription#getCenterLabelProvider
- EdgeDescription#getEndLabelProvider

to have something like that:

- EdgeDescription#getBeginLabelDescription: Optional<LabelDescription>
- EdgeDescription#getCenterLabelDescription: Optional<LabelDescription>
- EdgeDescription#getEndLabelDescription: Optional<LabelDescription>

Using the same LabelDescription as in Node.

It was tolerable when we were creating labels in the compatibility layer once, but we won't create all of them twice.